### PR TITLE
[DROOLS-7197] Generics work different in executable model than in MVEL

### PR DESCRIPTION
**Pull request by tkobayas**
- Cannot resolve the type in exec-model
- Added to KnownExecModelDifferenceTest
- Added docs

**JIRA**: 
https://issues.redhat.com/browse/DROOLS-7197

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] tests</b>

- for a <b>full downstream build</b> 
  - for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

- <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

- <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-main</b>

- for <b>quarkus lts checks</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins run quarkus-lts</b>

- for a <b>specific quarkus lts check</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-lts</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] mandrel</b>

 - for <b>mandrel lts checks</b>  
  Run native checks against Mandrel image and quarkus lts branch
  Please add comment: <b>Jenkins run mandrel-lts</b>

- for a <b>specific mandrel lts check</b>  
  Run native checks against Mandrel image and quarkus lts branch
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] mandrel-lts</b>
</details>

### CODE REVIEW

**Comment by kie-ci3** [comment]
**(tests) - kogito-quarkus-examples job** `#71` was: **UNSTABLE**
Possible explanation: This should be test failures

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-quarkus-examples/71/display/redirect

**Test results:**
- PASSED: 164
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-quarkus-examples/71/testReport/org.kie.kogito.quarkus.outbox/OutboxIT/___/">org.kie.kogito.quarkus.outbox.OutboxIT.(?)</a></summary>
java.lang.RuntimeException: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for container port to open (localhost ports: [8080, 49189] should be listening)<br/>	at org.kie.kogito.quarkus.outbox.OutboxIT.before(OutboxIT.java:86)<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for container port to open (localhost ports: [8080, 49189] should be listening)<br/>
</details>

**Comment by kie-ci3** [comment]
**(tests) - kogito-apps job** `#838` was: **UNSTABLE**
Possible explanation: This should be test failures

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-apps/838/display/redirect

**Test results:**
- PASSED: 1959
- FAILED: 2

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-apps/838/testReport/org.kie.kogito.index/ProcessDataIndexInfinispanIT/testProcessInstanceEvents/">org.kie.kogito.index.ProcessDataIndexInfinispanIT.testProcessInstanceEvents</a></summary>
java.lang.AssertionError: <br/>1 expectation failed.<br/>JSON path data.UserTaskInstances[0].attachments.size() doesn't match.<br/>Expected: is <1><br/>  Actual: <0><br/><br/>
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-apps/838/testReport/org.kie.kogito.index/ProcessDataIndexPostgreSqlIT/testProcessInstanceEvents/">org.kie.kogito.index.ProcessDataIndexPostgreSqlIT.testProcessInstanceEvents</a></summary>
java.lang.AssertionError: <br/>1 expectation failed.<br/>JSON path data.UserTaskInstances[0].attachments.size() doesn't match.<br/>Expected: is <1><br/>  Actual: <0><br/><br/>
</details>

**Comment by tkobayas** [comment]
GHA OptaPlanner : Not related to this PR
```
2022-12-15T09:39:44.9762862Z [ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:3.4.2:single (package-generated-docs) on project optaplanner-docs: Failed to create assembly: Error creating assembly archive zip-with-generated-docs: Problem creating zip: Execution exception: Java heap space -> [Help 1]
```

Jenkins kogito-apps : flaky tests
```
[2022-12-15T10:36:17.208Z] [ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 15.274 s <<< FAILURE! - in org.kie.kogito.index.ProcessDataIndexPostgreSqlIT
[2022-12-15T10:36:17.208Z] [ERROR] org.kie.kogito.index.ProcessDataIndexPostgreSqlIT.testProcessInstanceEvents  Time elapsed: 5.217 s  <<< FAILURE!
...
[2022-12-15T10:36:38.683Z] [ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 21.429 s <<< FAILURE! - in org.kie.kogito.index.ProcessDataIndexInfinispanIT
[2022-12-15T10:36:38.683Z] [ERROR] org.kie.kogito.index.ProcessDataIndexInfinispanIT.testProcessInstanceEvents  Time elapsed: 7.612 s  <<< FAILURE!
...

**Comment by tkobayas** [comment]
This issue falls into "Known difference" category because non-exec-model can dynamically resolve the type but exec-model cannot resolve the type (because of type erasure). So
- Add an explanation to "Migration from non-executable model to executable model" section
- Add test cases to KnownExecModelDifferenceTest

Please review, @mariofusco @pibizza @hellowdan @kaldesai , thanks!

File: drools-docs/src/modules/ROOT/pages/migration-guide/_nonexec-model-to-exec-model.adoc [hunk 1]
```diff
@@ -92,3 +92,66 @@ then
...
+----
+rule Rule1
+	when
+		$v : DieselCar(motor.adBlueRequired == true)
+	then
+		// do something
+end
+----
+
+`non-executable model` can dynamically resolve that `mortor` is `DieselEngine` so the rule works. However, `executable model` resolves  `mortor` to `TEngine`, so a build error is thrown.
```
Comment (kaldesai) [inline drools-docs/src/modules/ROOT/pages/migration-guide/_nonexec-model-to-exec-model.adoc]: I am assuming this is `motor`.

```suggestion
`non-executable model` can dynamically resolve that `motor` is `DieselEngine` so the rule works. However, the `executable model` resolves  `motor` to `TEngine`, so a build error is thrown.
```

File: drools-docs/src/modules/ROOT/pages/migration-guide/_nonexec-model-to-exec-model.adoc [hunk 2]
```diff
@@ -92,3 +92,66 @@ then
...
+
+----
+Unknown field adBlueRequired on TEngine
+----
+
+In this case, the rule can be fixed by specifying the subtype with `#` operator. For example,
```
Comment (kaldesai) [inline drools-docs/src/modules/ROOT/pages/migration-guide/_nonexec-model-to-exec-model.adoc]:
```suggestion
In this case, the rule can be fixed by specifying the subtype with the `#` operator. For example,
```

**Review by kaldesai — APPROVED** [review]
@tkobayas Minor changes. Otherwise, your content looks god to me. Thank you!

**Comment by tkobayas** [comment]
@kaldesai Thank you for the good catch!

**Comment by kie-ci3** [comment]
**(tests) - kogito-apps job** `#842` was: **UNSTABLE**
Possible explanation: This should be test failures

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-apps/842/display/redirect

**Test results:**
- PASSED: 1960
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-apps/842/testReport/org.kie.kogito.index/ProcessDataIndexPostgreSqlIT/testProcessInstanceEvents/">org.kie.kogito.index.ProcessDataIndexPostgreSqlIT.testProcessInstanceEvents</a></summary>
java.lang.AssertionError: <br/>1 expectation failed.<br/>JSON path data.UserTaskInstances[0].comments.size() doesn't match.<br/>Expected: is <1><br/>  Actual: <0><br/><br/>
</details>

**Review by mariofusco — APPROVED** [review]

**Pull request merged**